src/rate_limit.ts

```diff
@@ -230,20 +234,13 @@ export default class RateLimitPlugin {
             try {
               innerSpan.setAttribute("internal.visibility", String("user"));
               innerSpan.setAttribute("rate_limit.key", key);
-              const limit =
-                this.config.rate_limit.role_based_limits.find(
-                  (limit) => limit.role === request.session.role,
-                )?.limit || this.config.rate_limit.default_limit;
```

Comment:
We were not using this limit, this PR just fixes the code to use this limit instead of default limit.